src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -196,6 +199,12 @@ public void start(Map<String, String> properties) {
     TimeZone timeZone = config.timeZone();
     String suffix = config.getString(JdbcSourceTaskConfig.QUERY_SUFFIX_CONFIG).trim();
 
+    final List<String> tableTypes = config.getList(JdbcSourceConnectorConfig.TABLE_TYPE_CONFIG);
+    final Connection conn = cachedConnectionProvider.getConnection();
+    if (tableTypes.contains(TABLE_TYPE_SYNONYM) && conn instanceof OracleConnection) {
+      ((OracleConnection) conn).setIncludeSynonyms(true);
```

Comment:
Also, Please avoid adding DB specific code in common Class. We have set of Dialect classes which can be used in such case.